### DISCUSSION TITLE
Hide window's app instead of minimising window on click

### DISCUSSION THREAD

**Discussion by Vybo** [lawand-dot-io/taskbar#7] [Feature requests]
Currently, clicking on an app in Taskbar will minimize the app/window to the dock (with the corresponding animation). It would be nice to have an option which would hide the window instead of minimizing it.

**Comment by lawand-dot-io** [comment]
As far as I understand, the OS only supports hiding whole Apps and not individual windows. I will look into whether there's a way to get around this limitation, but if I don't find anything then you're stuck with only minimizing them

**Reply by Vybo** [reply]
Even just hiding all app windows would be acceptable from my perspective, if it was configurable, thanks 👍

**Reply by lawand-dot-io** [reply]
Interesting idea. For me personally I only want to minimize one window when I click on it, but if you prefer hiding the app, I can surely implement this. Should be easy. This will be an option in the next version :)

**Reply by jameszhan9592** [reply]
Is it possible to find a way to invoke the CMD + H shortcut on the specific app window when you click on it on Taskbar? I'm not a programmer at all but I would also prefer hiding the app window over minimizing.

**Reply by lawand-dot-io** [reply]
@jameszhan9592 What's that advantage of hiding and not minimizing? :)

**Reply by jameszhan9592** [reply]
@lawand-dot-io The advantages of hiding vs minimizing are:

1. Reduced animation. When a window is being minimized, you are stuck with either the scaled effect or the genie effect, but when you do CMD + H, it's instant. It feels a lot snappier and less distracting
2. With Taskbar running, I've found a new usage for Mac's Dock. I would occasionally use it to launch apps that I use less frequently, and I don't pin these apps in Taskbar to reduce visual clutter. When I have a lot of windows minimized, they take up a lot of space in the Dock and they make the Dock look really small (because the Dock has to resize due to a lack of space).

**Reply by livinlicious** [reply]
I want to add to this, that I would appreciate this feature as well.
Its acceptable if clicking hides ALL instances of this App. (as long as there is a toggle checkbox in the settings for this / in general everything should be optional, everyone works differently)

**Reply by lawand-dot-io** [reply]
@livinlicious sure

**Comment by lawand-dot-io** [comment]
This feature has been implemented in Taskbar version 1.2.1 which is now released. Closing the ticket.

@Vybo
